packages/e2e-tests/specs/performance/front-end-block-theme.test.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * External dependencies
+ */
+import { basename, join } from 'path';
+import { writeFileSync } from 'fs';
+
+/**
+ * WordPress dependencies
+ */
+import { activateTheme, createURL, logout } from '@wordpress/e2e-test-utils';
+
+describe( 'Front End Performance', () => {
+	const results = {
+		timeToFirstByte: [],
+	};
+
+	beforeAll( async () => {
+		await activateTheme( 'twentytwentythree' );
+		await logout();
+	} );
+
+	afterAll( async () => {
+		await activateTheme( 'twentytwentyone' );
+		const resultsFilename = basename( __filename, '.js' ) + '.results.json';
+		writeFileSync(
+			join( __dirname, resultsFilename ),
+			JSON.stringify( results, null, 2 )
+		);
+	} );
+
+	it( 'Time To First Byte (TTFB)', async () => {
+		// We derive the 75th percentile of the TTFB based on these results.
+		// By running it 16 times, the percentile value would be (75/100)*16=12,
+		// meaning that we discard the worst 4 values.
+		let i = 16;
+		while ( i-- ) {
+			await page.goto( createURL( '/' ) );
+			const navigationTimingJson = await page.evaluate( () =>
+				JSON.stringify( performance.getEntriesByType( 'navigation' ) )
+			);
+			const [ navigationTiming ] = JSON.parse( navigationTimingJson );
```

Comment:
Nit:

`evaluate` should automatically serialize/deserialize JSON data, so we should be able to just do:

```js
const [ navigationTimeing ] = await page.evaluate( () =>
  performance.getEntriesByType( 'navigation' )
);
```

In addition, we already have some performance utils available, and this probably belongs in the `getLoadingDurations` util. WDYT?

Reply:
https://github.com/WordPress/gutenberg/pull/47541